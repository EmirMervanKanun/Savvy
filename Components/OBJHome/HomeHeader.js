import { StyleSheet, View, Image, Text } from "react-native";
import COLORS from '../Farben';
import Header from '../TextComponents/Header';
import PlaceholderText from '../TextComponents/PlaceholderText';
import ButtonSettings from "../Buttons/ButtonSettings";

const HomeHeader = ({ props }) => {
    return (
        <View style={styles.container}>
            <View style={styles.head}>
                <View style={styles.personalData}>
                    <Image source={props.profilePicture} style={styles.imagePerson} />
                    <Text style={styles.name}><Header>Hallo, {props.name}!</Header></Text>
                </View>
                <ButtonSettings/>

            </View>
            <View style={styles.money}>
                <Text style={{ color: COLORS.schriftLight }}><Header>{props.amount}â‚¬</Header></Text>
                <Text style={{ color: COLORS.schriftMid }}><PlaceholderText>Gesamtsaldo</PlaceholderText></Text>
            </View>
        </View>
    );
}
export default HomeHeader;

const styles = StyleSheet.create({
    container: {
        height: 184,
        width: '100%',
        backgroundColor: COLORS.primaryDark,
        borderBottomRightRadius: 70,
        paddingTop: 20,
    },
    imagePerson: {
        width: 70,
        height: 70,
    },
    head: {
        display: 'flex',
        flexDirection: 'row',
        justifyContent: "space-between",
        alignItems: 'center',
        paddingHorizontal: 40,
    },
    personalData: {
        display: 'flex',
        flexDirection: 'row',
        gap: 10,
        alignItems: 'center',
    },
    name: {
        color: COLORS.schriftLight,
        width: '60%',
    },
    money: {
        flexDirection: 'column',
        alignItems: 'flex-start',
        paddingTop: 20,
        paddingLeft: 40,
    },
});