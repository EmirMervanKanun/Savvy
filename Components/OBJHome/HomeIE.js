import { StyleSheet, View, Image, Text } from "react-native";
import COLORS from '../Farben';
import TitleAmountText from '../TextComponents/TitelAmountText';
import NavText from '../TextComponents/NavText';
import DateText from '../TextComponents/DateText';

//I = Income, E = Expenses

const HomeIE = ({ props }) => {
    return (
        <View style={styles.container}>
            <View style={[styles.income, styles.box]}>
                <View style={styles.text}>
                    <Text style={{ color: COLORS.schriftMid }}><NavText>Einnahmen</NavText></Text>
                    <Text style={{ color: COLORS.schriftDark }}><DateText>April</DateText></Text>
                </View>
                <View style={styles.bildmittext}>
                    <Image source={require('../../Icons/Objects/arrowIncome.png')} style={styles.image} />
                    <Text style={{ color: COLORS.greenDark }}><TitleAmountText>{props.iAmount}€</TitleAmountText></Text>
                </View>
            </View>
            <View style={[styles.expenses, styles.box]}>
                <View style={styles.text}>
                    <Text style={{ color: COLORS.schriftMid }}><NavText>Ausgaben</NavText></Text>
                    <Text style={{ color: COLORS.schriftDark }}><DateText>April</DateText></Text>
                </View>
                <View style={styles.bildmittext}>
                    <Image source={require('../../Icons/Objects/arrowExpense.png')} style={styles.image} />
                    <Text style={{ color: COLORS.redDark }}><TitleAmountText>- {props.eAmount}€</TitleAmountText></Text>
                </View>
            </View>
        </View>
    );
}
export default HomeIE;

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        gap: 20,
    },
    box: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        height: 104,
        width: 136,
        borderRadius: 15,
        gap: 4,
    },
    income: {
        backgroundColor: COLORS.greenLight,
    },
    expenses: {
        backgroundColor: COLORS.redLight,
    },
    text: {
        flexDirection: 'column',
        alignItems: 'center',
    },
    bildmittext: {
        flexDirection: 'row',
        gap: 8,
        alignItems: 'center',
    },
    image: {
        width: 32,
        height: 32,
    },
});