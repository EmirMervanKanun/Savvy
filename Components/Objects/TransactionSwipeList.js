import React, { useState } from "react";
import { View, StyleSheet, TouchableHighlight } from "react-native";
import { SwipeListView } from "react-native-swipe-list-view";
import TransStanOrder from "./TransStanOrder";
import Button from "../Buttons/Button";

const TransactionSwipeList = ({ transactions }) => {
    const [listData, setListData] = useState(
        transactions.map((TransactionItem, index) => ({
            key: `${index}`,
            title: TransactionItem.title,
            details: TransactionItem.details,
        }))
    );

    const VisibleItem = props => {
        const { data } = props;
        return (
            <TouchableHighlight>
                <TransStanOrder props={data.item} />
            </TouchableHighlight>
        );
    };

    const renderItem = (data) => {
        return (
            <View style={styles.itemContainer}>
                <VisibleItem data={data} />
            </View>
        );
    };

    const renderHiddenItem = () => {
        return (
            <View style={styles.rowBack}>
                <View style={[styles.backRightBtn, styles.backRightBtnLeft]}>
                    <Button
                        props={{ 
                            color: "blue", 
                            img: require("../../Icons/Button/edit.png"),
                            onPress: () => console.log("Edit button clicked")
                        }}
                    />
                </View>

                <View style={[styles.backRightBtn, styles.backRightBtnRight]}>
                    <Button
                        props={{ 
                            color: "red", 
                            img: require("../../Icons/Button/delete.png"), 
                            onPress: () => console.log("Delete button clicked")
                        }}
                    />
                </View>
            </View>
        );
    };

    return (
        <SwipeListView
            contentContainerStyle={styles.container}
            data={listData}
            renderItem={renderItem}
            renderHiddenItem={renderHiddenItem}
            rightOpenValue={-100}
            disableRightSwipe
        />
    );
};

const styles = StyleSheet.create({
    container: {
        display: "flex",
    },
    itemContainer: {
        marginBottom: 24,
        backgroundColor: "white",
    },
    rowBack: {
        alignItems: 'center',
        backgroundColor: 'white',
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'flex-end',
    },
    backRightBtn: {
        alignItems: 'center',
        justifyContent: 'center',
        position: 'absolute',
        top: 8.5,
    },
    backRightBtnLeft: {
        right: 50,
    },
    backRightBtnRight: {
        right: 0,
    },
});

export default TransactionSwipeList;
