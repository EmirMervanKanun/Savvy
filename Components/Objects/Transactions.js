const Categories = [
    {
        title: "Transaction1",
        details: {
            isIncome: false,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: false,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: false,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: false,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    {
        title: "Transaction1",
        details: {
            isIncome: true,
            img: require('../../Icons/Categories/gaming.png'),
            amount: 1000,
            date: "01.01.2021",
            isStan: true,
            stan: "Monthly",
        }
    },
    
];

export default Categories;