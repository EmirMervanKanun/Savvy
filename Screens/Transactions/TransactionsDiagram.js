import { View, Text } from 'react-native';
import React from 'react';


export default function TransactionsDiagramm() {
  return (
    <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
      <Text>Transaktionen Diagramm</Text>
    </View>
  );
}