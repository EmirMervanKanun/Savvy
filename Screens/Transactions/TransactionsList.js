import { View, StyleSheet, ScrollView } from 'react-native';
import React from 'react';
import TransactionsSwipeList from '../../Components/Objects/TransactionSwipeList';
import ButtonFilter from '../../Components/Buttons/ButtonFilter';
import Transactions from '../../Components/Objects/Transactions';

let filterSelected;
function TransactionsList() {
  return (
    <ScrollView >
      <View style={styles.container}>
        {//Monatsswitcher
        }
        <ButtonFilter options={["Gesamt", "Einnahmen", "Ausgaben"]} chnagedilter={filterSelected} selectedAtStart={"Gesamt"}/>
        <View style={styles.budgets}>
          <TransactionsSwipeList transactions={Transactions} />
        </View>
      </View>
    </ScrollView>
  );
}


export default function TransactionScreen() {
  return (
    <TransactionsList/>
  );
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: 'white',
    paddingTop: 24,
    paddingBottom: 220,
    gap: 32,
  },
  budgets: {
    paddingTop: 8,
    display: 'flex',
    flexDirection: 'column',
  },
});