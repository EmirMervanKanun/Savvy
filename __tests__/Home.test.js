import React from 'react';
import renderer, { act } from 'react-test-renderer';
import HomeStart from '../Screens/Home/Home';
import HomeHeader from '../Components/OBJHome/HomeHeader';
import HomeIE from '../Components/OBJHome/HomeIE';
import TransactionSwipeList from '../Components/Objects/TransactionSwipeList';
import Transactions from '../Components/Objects/Transactions';

// Mock der untergeordneten Komponenten
jest.mock('../Components/OBJHome/HomeHeader', () => 'HomeHeader');
jest.mock('../Components/OBJHome/HomeIE', () => 'HomeIE');
jest.mock('../Components/Objects/TransactionSwipeList', () => 'TransactionSwipeList');
jest.mock('../Components/Objects/Transactions', () => [
  {
    id: 1,
    type: 'expense',
    amount: '50.00',
    date: '2024-07-10',
  },
  {
    id: 2,
    type: 'income',
    amount: '100.00',
    date: '2024-07-09',
  },
]);

describe('HomeStart Component', () => {
  it('renders correctly', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<HomeStart />);
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders HomeHeader with correct props', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<HomeStart />);
    });

    const homeHeaderInstance = tree.root.findByType(HomeHeader);
    expect(homeHeaderInstance.props.props.profilePicture).toBe(
      require('../Icons/ProfilePictures/profilePicture1.png')
    );
    expect(homeHeaderInstance.props.props.name).toBe('John');
    expect(homeHeaderInstance.props.props.amount).toBe('123.456,89');
  });

  it('renders HomeIE with correct props', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<HomeStart />);
    });

    const homeIEInstance = tree.root.findByType(HomeIE);
    expect(homeIEInstance.props.props.iAmount).toBe('1.234,56');
    expect(homeIEInstance.props.props.eAmount).toBe('7.890,12');
  });

  it('renders TransactionSwipeList with correct transactions', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<HomeStart />);
    });

    const transactionSwipeListInstance = tree.root.findByType(TransactionSwipeList);
    expect(transactionSwipeListInstance.props.transactions).toEqual(Transactions);
    expect(transactionSwipeListInstance.props.height).toBe(264);
  });
});
