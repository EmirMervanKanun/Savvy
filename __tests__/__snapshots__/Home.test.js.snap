// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeStart Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "display": "flex",
      "gap": 20,
      "height": "100%",
    }
  }
>
  <HomeHeader
    props={
      {
        "amount": "123.456,89",
        "name": "John",
        "profilePicture": 1,
      }
    }
  />
  <HomeIE
    props={
      {
        "eAmount": "7.890,12",
        "iAmount": "1.234,56",
      }
    }
  />
  <View>
    <Text
      style={
        {
          "fontFamily": "Sora-Regular",
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "Sora-Bold",
            "fontSize": 20,
            "lineHeight": 24,
          }
        }
      >
        Transaktionen
      </Text>
    </Text>
    <TransactionSwipeList
      height={264}
      transactions={
        [
          {
            "amount": "50.00",
            "date": "2024-07-10",
            "id": 1,
            "type": "expense",
          },
          {
            "amount": "100.00",
            "date": "2024-07-09",
            "id": 2,
            "type": "income",
          },
        ]
      }
    />
  </View>
</View>
`;
